### DISCUSSION TITLE
Conda and reproducibility

### DISCUSSION THREAD

**Discussion by s-canchi** [nih-cfde/training-and-engagement#371] [Conda Q&A] [answered]
What does it mean to be OS agnostic? Thinking about reproducibility, if I install some software using conda on Mac OS, do I get the same software as someone who uses my environment definition (e.g. with a yaml file) on linux or windows?

**Comment by s-canchi** [comment, marked as answer]
- OS agnostic means that conda works on Windows, Mac, and Linux!
- Yes, the software would be the same as long as the same software packages are specified. You can also have them replicate the exact environment. To do so, access the conda environment you wish to replicate:
```
conda activate env
conda list --export > packages.txt
conda deactivate
```
Two options to replicate the environment:

1) install the exact package list into an existing environment:

```
conda install --file=packages.txt
```

2) set up a new environment with the exact package list:
```
conda env create --name qc_file --file packages.txt
```

**Reply by ctb** [reply]
a few extra thoughts --

conda works on Windows, Mac, and Linux! However, specific packages may not be available on all of those. For example, I believe both conda-forge and bioconda generally provide Mac OS and Linux packages, but not necessarily Windows packages.

an important addendum to the above is that for bitwise-identical packages, you will want to use the `conda list --export` command above, which will give you not only the same version of the software package but the same build. However, that is not cross platform compatible. So it is limiting to do things this way and generally you should specify the version but not the exact package list.
